app/e2e/outgoingcall.test.js

```diff
@@ -0,0 +1,82 @@
+
+describe('Outgoing Call', () => {
+  beforeAll(async () => {
+    await global.device.launchApp();
+  });
+
+  const navigateToDialer = async () => {
+    await element(by.id('login_button')).tap();
+    await element(by.id('dialer_button')).tap();
+  };
+
+  beforeEach(async () => {
+    await global.device.reloadReactNative();
+    await navigateToDialer();
+  });
+
+  it('should allow user to enter PTSN number', async () => {
+    await element(by.id('dialpad_button_1')).tap();
+    await element(by.id('dialpad_button_2')).tap();
+    await element(by.id('dialpad_button_3')).tap();
+    await element(by.id('dialpad_button_4')).tap();
+    await element(by.id('dialpad_button_4')).tap();
+    await element(by.id('dialpad_button_4')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await expect(element(by.id('formatted_number'))).toHaveText('+1234448888');
+  });
+
+  it('should allow user to enter Client-ID', async () => {
+    await element(by.text('Client')).tap();
+    await element(by.id('client_text_input')).typeText('Client-Web');
+    await expect(element(by.id('client_text_input'))).toHaveText('Client-Web');
+  });
+
+  if (device.getPlatform() === 'android') {
+    describe('Android: successful connect', () => {
+      it('should make a successful PTSN call', async () => {
+        await element(by.id('dialpad_button_3')).tap();
+        await element(by.id('dialpad_button_1')).tap();
+        await element(by.id('dialpad_button_5')).tap();
+        await element(by.id('dialpad_button_6')).tap();
+        await element(by.id('dialpad_button_7')).tap();
+        await element(by.id('dialpad_button_0')).tap();
+        await element(by.id('dialpad_button_3')).tap();
+        await element(by.id('dialpad_button_5')).tap();
+        await element(by.id('dialpad_button_4')).tap();
+        await element(by.id('dialpad_button_1')).tap();
+        await element(by.id('call_button')).tap();
+        await waitFor(element(by.id('active_call'))).toBeVisible();
+        await waitFor(element(by.id('call_status'))).toHaveText('ringing');
+        await waitFor(element(by.id('call_status'))).toHaveText('00:05');
+        await waitFor(element(by.id('active_call'))).not.toBeVisible();
+      });
+    });
+
+    describe('disconnect', () => {
+      it('should disconnect if invalid number', async () => {
+        await element(by.id('dialpad_button_1')).tap();
+        await element(by.id('dialpad_button_2')).tap();
+        await element(by.id('dialpad_button_3')).tap();
+        await element(by.id('call_button')).tap();
+        await waitFor(element(by.id('active_call'))).toBeVisible();
+        await waitFor(element(by.id('call_status'))).toHaveText('ringing');
+        await waitFor(element(by.id('call_status'))).toHaveText('disconnected');
```

Comment:
Can you also check that the call gets disconnected right away? It should get disconnected in less than 3s, is that correct?

Reply:
I don't think it can be right away, since the UI still needs time to switch Views. That is why I used a waitFor, since without it the test fails. Where does the 3s come from?

Reply:
By "right away" I mean, it should not feel like it's stuck. 3s is just an estimate. What we want is, we need to fail this test if it exceeds a certain threshold. For example, if it does not disconnect after 20s, there must be something wrong. 20s is too extreme, so I estimated around 3s. How long do you think we should wait before we consider this as a fail?

Reply:
I get what you mean now. Okay will add another testcase. 3s sounds like a good time to me 👍